server/controllers/flightRecord.js

```diff
@@ -16,7 +16,8 @@ const addFlightRecord = async (req, res, next) => {
 
 const getFlightRecords = async (req, res, next) => {
   try {
-    const records = await findFlightRecords();
+    let records = await findFlightRecords(req.query);
+    records = records.filter((record) => record.flightId);
```

Comment:
it's better to do with DB filter (not array filter). It will help you in case of pagination